src/lib/Libifl/int_status.c

```diff
@@ -46,7 +46,132 @@
 
 #include <string.h>
 #include <stdio.h>
+#include <ctype.h>
 #include "libpbs.h"
+#include "pbs_ecl.h"
+#include "libutil.h"
+#include "attribute.h"
+#include "cmds.h"
+#include "pbs_internal.h"
+
+
+extern char * PBS_get_server(char *, char *, uint *);
+
+
+/**
+ * @brief
+ * 	get_available_conn:
+ *	get one of the available connection from multisvr sd
+ *
+ * @param[in] svr_connections - pointer to array of server connections
+ * 
+ * @return int
+ * @retval -1: error
+ * @retval != -1 fd corresponding to the connection
+ */
+int
+get_available_conn(svr_conn_t *svr_connections)
+{
+	int i;
+
+	for (i = 0; i < get_num_servers(); i++)
+		if (svr_connections[i].state == SVR_CONN_STATE_UP)
+			return svr_connections[i].sd;
+
+	return -1;
+}
+
+/**
+ * @brief
+ * random_srv_conn:
+ *	get random server sd - It will choose a random sd from available no of servers.
+ * @param[in] svr_connections - pointer to array of server connections
+ * 
+ * @return int
+ * @retval -1: error
+ * @retval != -1: fd corresponding to the connection
+ */
+int
+random_srv_conn(svr_conn_t *svr_connections)
+{
+	int ind = 0;
+
+	ind =  rand_num() % get_num_servers();
+
+	if (svr_connections[ind].state == SVR_CONN_STATE_UP)
+		return svr_connections[ind].sd;
+		
+	return get_available_conn(svr_connections);
```

Comment:
When we reach here which means the chosen random server is down. So when we call the function get_available_conn() we should only look a random server from the remaining servers right ?. Here it seems again we are looping through all the available servers inside get_available_conn.

Reply:
It does not worth saving the failed connection as the cost we are saving will be spent in the form of an if condition.

Reply:
Why can't we simply mark the connection as SVR_CONN_STATE_DOWN so that this particular connection is not exercised again in get_available_conn. If we do this then we no need to change the code of get_available_conn.

Reply:
Can you clarify a bit more? This function will find a random fd. If that fd is not connected then it will loop and get an available connection. That fd would have already marked as down. Otherwise, the function will return the same fd from here.